packages/blockchain-wallet-v4-frontend/src/scenes/RecoverWallet/ResetAccount/TwoFAConfirmation/index.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from 'react'
+import { FormattedMessage } from 'react-intl'
+import { Field } from 'redux-form'
+import styled from 'styled-components'
+
+import { HeartbeatLoader, Text } from 'blockchain-info-components'
+import Form from 'components/Form/Form'
+import FormError from 'components/Form/FormError'
+import FormGroup from 'components/Form/FormGroup'
+import FormItem from 'components/Form/FormItem'
+import FormLabel from 'components/Form/FormLabel'
+import PasswordBox from 'components/Form/PasswordBox'
+import TextBox from 'components/Form/TextBox'
+import { selectors } from 'data'
+import { ModalName } from 'data/modals/types'
+import { RecoverSteps } from 'data/types'
+import { useRemote } from 'hooks'
+import { required } from 'services/forms'
+import { removeWhitespace } from 'services/forms/normalizers'
+import { isMobile, media } from 'services/styles'
+
+import { Props } from '../..'
+import { ActionButton, BackArrowFormHeader, CenteredColumn, FormWrapper, Row } from '../../model'
+
+const ResponsiveRow = styled(Row)`
+  justify-content: center;
+  margin-top: 16px;
+  ${media.mobile`
+    flex-direction: column;
+    align-items: center;
+    line-height: 2;
+  `}
+`
+
+const TwoFAConfirmation = (props: Props) => {
+  const { hasError } = useRemote(selectors.signup.getRecoveryTwoFAVerification)
+  const { accountRecoveryData, busy, formValues, invalid, setStep, signupActions, submitting } =
+    props
+  const authType = accountRecoveryData && accountRecoveryData?.two_fa_type
+  const placeholder = authType === 4 ? '_ _ _  _ _ _' : authType === 5 ? '_ _ _ _ _' : null
+  // authType 5 is mobile sms code, 5 characters
+  // authType 4 is OTP code, 6. yubikey is at least 6
+  const minRequiredCharaters = authType === 4 ? 6 : 5
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    signupActions.verifyTwoFaForRecovery({
+      code: formValues?.twoFACode,
+      email: formValues.recoveryEmail
+    })
+  }
+
+  //   const accountLocked =
```

Comment:
can likely remove these, have a todo to make sure i'm handling incorrect 2fa correctly